src/partial-templates/partial-templates.service.ts

```diff
@@ -16,33 +16,49 @@ export class PartialTemplatesService {
    * @param {CreatePartialTemplateDto} createPartialTemplateDto - The partialTemplate data to register
    * @returns {Promise<PartialTemplate>} - The registered partialTemplate
    */
-  async create(createPartialTemplateDto: CreatePartialTemplateDto): Promise <PartialTemplate> {
-    // Checks if a partial template with the same name already exists
-    const existingPartialTemplate =  await this.prisma.partialTemplate.findFirst({
-      where: {
-        name: createPartialTemplateDto.name,
-      },
-    });
-    // Throws an exception if a template with the same name exists
-    if (existingPartialTemplate){
-      throw new BadRequestException('A partial template with this name already exists');
-    }
-
-    // Creates a new partial template
-    return this.prisma.partialTemplate.create({
-      data: {
-        ...createPartialTemplateDto,
-      } as any,
-    });
-  }
+    async create(createPartialTemplateDto: CreatePartialTemplateDto): Promise <{ message: string | null; error: { message: string } | null; data: PartialTemplate | null }> {
+      try {
+      await this.validateForeignKeys(createPartialTemplateDto);
+      await this.validateTotalByPosition(createPartialTemplateDto);
+      const newPartialTemplate = await this.prisma.partialTemplate.create({
+        data: {
+          ...createPartialTemplateDto,
+        },
+      });
+  
+      return { message: 'Listo, enviado', error: null, data: newPartialTemplate};
+    } catch (error) {
+      return { message: 'Error al enviar', error: error.message , data: null };
+    }
+ }
 
```

Comment:
### In all CRUD methods upgrade to a Try-Catch block to ensure that errors are handled properly, which improves the robustness and reliability of the code.

- **Example:**
```ts
 async create() {
 try {
     // Main action performed in the try block
     // Return responses
     return {
         message: 'Operation successful',
         error: null,
         data: result,
      };
  } catch (error) {
      //Handle any errors that occur during the operation
      return {
          message: 'Error performing operation',
          error: error.message,
          data: null,
     };
  }
}
```